### ISSUE TITLE
Extension host restarts all the time

### ISSUE THREAD

**Issue by codefriar** [microsoft/vscode#176135]
Type: <b>Bug</b>

Not sure how to debug what extension(s) is causing this, but the extension host keeps crashing.

VS Code version: Code 1.76.0 (92da9481c0904c6adfe372c12da3b7748d74bdcb, 2023-03-01T10:23:06.752Z)
OS version: Darwin arm64 22.3.0
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (8 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|6, 6, 6|
|Memory (System)|32.00GB (0.73GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (95)</summary>

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|3.0.2
rubocop-quick-fix|aki|0.2.0
project-manager|ale|12.7.0
jest-snippets|and|1.9.1
quitcontrol-vscode|art|4.0.0
better-toml|bun|0.3.2
solargraph|cas|0.24.0
apex-pmd|chu|0.5.9
codefriar-s-pack|cod|2.0.0
erb|Cra|0.0.1
vscode-eslint|dba|2.4.0
vscode-npm-script|eg2|0.3.29
RunOnSave|eme|0.2.0
prettier-vscode|esb|9.10.4
comment-anchors|Exo|1.10.3
lana|Fin|1.5.2
auto-close-tag|for|0.5.14
auto-rename-tag|for|0.1.10
code-runner|for|0.12.0
copilot|Git|1.77.9225
vscode-pull-request-github|Git|0.58.2
gitlab-workflow|Git|3.60.0
go|gol|0.38.0
ruby|gro|0.1.0
todo-tree|Gru|0.0.224
ruby-linter|hoo|1.0.0
rest-client|hum|0.25.1
output-colorizer|IBM|0.1.2
plantuml|jeb|2.17.5
vscode-todo-highlight|jgc|2.0.4
dupchecker|jia|0.1.7
gitlab-notifications|log|0.0.8
gitless|maa|11.7.2
gitignore|mic|1.0.1
ruby-symbols|mig|0.1.8
sfdcloganalysis|Mik|0.0.7
dotenv|mik|1.0.1
ruby-rubocop|mis|0.8.6
apex-code-coverage-visualizer|mod|0.9.8
better-folding|Moh|0.5.1
vscode-json5|mrm|1.0.0
vscode-docker|ms-|1.24.0
jupyter-keymap|ms-|1.1.0
remote-containers|ms-|0.282.0
remote-ssh|ms-|0.98.0
remote-ssh-edit|ms-|0.84.0
remote-wsl|ms-|0.76.1
vscode-remote-extensionpack|ms-|0.24.0
azure-account|ms-|0.11.3
live-server|ms-|0.4.6
makefile-tools|ms-|0.6.0
remote-explorer|ms-|0.2.0
remote-server|ms-|1.0.0
vscode-typescript-tslint-plugin|ms-|1.3.4
vsliveshare|ms-|1.0.5832
dataweave|Mul|1.2.15
vscode-paste-image|mus|1.0.4
indent-rainbow|ode|8.3.1
apexdox-vs-code|Pet|1.5.1
jq-vscode|pet|0.1.3
material-icon-theme|PKi|4.24.0
heroku-command|pko|0.0.8
polacode|pnp|0.3.4
sourcekit-lsp--dev-unofficial|pva|0.20190711.1
ruby|reb|0.28.1
vscode-commons|red|0.0.6
vscode-xml|red|0.24.0
vscode-yaml|red|1.11.0
salesforce-vscode-slds|sal|1.4.8
salesforcedx-vscode|sal|57.3.0
salesforcedx-vscode-apex|sal|57.3.0
salesforcedx-vscode-apex-debugger|sal|57.3.0
salesforcedx-vscode-core|sal|57.3.0
salesforcedx-vscode-expanded|sal|57.3.0
salesforcedx-vscode-lightning|sal|57.3.0
salesforcedx-vscode-lwc|sal|57.3.0
salesforcedx-vscode-soql|sal|57.3.0
salesforcedx-vscode-visualforce|sal|57.3.0
gitconfig|sid|2.0.1
indenticator|Sir|0.7.0
swdc-vscode|sof|2.6.48
code-spell-checker|str|2.18.0
html-preview-vscode|tht|0.2.5
errorlens|use|3.7.0
reopenclosedtab|uyi|1.1.0
sfdx-package-xml-generator|Vig|2.0.9
intellicode-api-usage-examples|Vis|0.2.7
vscodeintellicode|Vis|1.2.30
simple-ruby-erb|vor|0.2.1
gistfs|vsl|0.4.1
Ruby-extension-pack|wal|1.0.0
vscode-ruby|win|0.28.0
JavaScriptSnippets|xab|1.8.0
markdown-all-in-one|yzh|3.5.0
html-css-class-completion|Zig|1.20.0

(2 theme extensions excluded)

</details>

**Comment by codefriar** [comment]
If there's a 'figure out what extension is causing you heartache' guide my google-fu is failing me. If you want to link me i'll be happy to track this down as far as I'm able.

**Comment by roblourens** [comment]
The "Start Extension Bisect" command could help if this issue is easily reproducible

**Comment by Croydon-Brixton** [comment]
I'm experiencing the same issue since today, have not found out why yet.

**Comment by bpasero** [comment]
Can you please follow the steps in https://github.com/Microsoft/vscode/wiki/Native-Crash-Issues to get at more details around the crash and attach the result here? Thanks!

**Comment by Croydon-Brixton** [comment]
Hi, thank you for your help!

I followed the instructions under the link you provided, and here's what I get. Hope this is helpful, this is the first time I am doing this, so let me know if there should be any issue.  

[08c82151-80b8-4939-b419-47783f6ef981.dmp.zip](https://github.com/microsoft/vscode/files/10930279/08c82151-80b8-4939-b419-47783f6ef981.dmp.zip)

**Comment by Croydon-Brixton** [comment]
Yes, unfortunately the same problem happens when running with `--no-sandbox`. 

The crashing behavior is somewhat stochastic, I can usually run a few lines of code in a jupyter notebook until the notebooks window suddenly closes and fails to save (the overall VSCode window remains open and functional) 

Here's the dump from running in `--no-sandbox` mode:
[116976c0-55dc-4336-a3ba-563efd30d8a7.dmp.zip](https://github.com/microsoft/vscode/files/10931869/116976c0-55dc-4336-a3ba-563efd30d8a7.dmp.zip)

**Comment by Croydon-Brixton** [comment]
Could this be related to https://github.com/microsoft/vscode/issues/176562?

**Comment by deepak1556** [comment]
@Croydon-Brixton the crash logs you had shared have different root issue from https://github.com/microsoft/vscode/issues/176562. To clarify, do you also have copilot extension installed ?

**Comment by Croydon-Brixton** [comment]
Thank you, yes I also have the copilot extension installed and am currently testing whether the crash occurs if I deactivate copilot. Will update you as I learn more.
